src/io/GPIO.cpp

```diff
@@ -0,0 +1,34 @@
+#include <EVT/io/GPIO.hpp>
+
+// Find out which platform being built for
+#ifdef STM32F302x8
+    #include <EVT/io/platform/f3xx/GPIOf3.hpp>
+#else
+    #error "The target platform does not support GPIO"
+#endif
+
+namespace EVT::core::IO
+{
+
+// Setup instance variables
+GPIO::GPIO(Pin pin) 
+{
+    this->pin = pin;
+}
+
+GPIO::GPIO(Pin pin, Direction direction) 
+{
+    this->pin = pin;
+    this->direction = direction;
+}
+
+GPIO& GPIO::getInstance(Pin pin, Direction direction)
+{
+    // Determine which GPIO interface to use
+    #ifdef STM32F302x8
+        static GPIOf3 gpioPin(pin, direction);
```

Comment:
I'm not totally a fan of labelling the GPIO adapter as just "F3", you're moving from less to more abstract here. The handlers written for it are specific to "F302x8", and may not necessarily apply to all F3 models. Same applies for the other future IO modules.

Reply:
I think for right now we should be ok, the code in the GPIOf3 adapter does not necessarily make use of F302x8 specific code. In the future if we adopt more F3 models, then I think we would extend GPIOf3 to have GPIOF302x8 etc.

Reply:
You may be okay with GPIO because it is so trivial, but I can see other peripherals on other F3 models where you could run into trouble here. The IRQ handlers are defined initially in `startup_stm32f302x8.s` and their layout is most definitely model-specific. And while the handler names may be the same from model to model, it's not guaranteed that their implementation will be that way. I hate to hold up a PR over a semantics argument but if we're going through all the trouble to support multiple controller families we might as well be this explicit from the get go.

Reply:
Yeah that's a good point, I added a sub-folder to f3xx that has the f302x8 code renamed within. In the future if we expand to include more f3 models they can go into the f3xx folder.